### ISSUE TITLE
Wrong DataPalette.readLegacy 

### ISSUE THREAD

**Issue by caoli5288** [retrooper/packetevents#472]
**Describe the bug**
In the following code sometimes negative numbers are read, causing an unusually large array to be allocated in the ListPalette.

```
    public static DataPalette readLegacy(NetStreamInput in) {
        int bitsPerEntry = in.readByte();
```

**Software brand**
paper-1.12.2

**Plugins**
Plugins (1): GrimAC

**How To Reproduce**
1. Put an legacy world in server
2. start server
3. Player join

![图片](https://user-images.githubusercontent.com/3120587/186160053-7ba8df3e-3505-4f80-8242-d1cef5eace8e.png)

![图片](https://user-images.githubusercontent.com/3120587/186160128-810fdc98-4b2d-4896-8815-3b3ed0886956.png)

**Comment by caoli5288** [comment]
Related to https://github.com/GrimAnticheat/Grim/issues/729

**Comment by caoli5288** [comment]
![186144498-e295eed6-ef47-4d85-ac33-9f4761b9c011](https://user-images.githubusercontent.com/3120587/186160789-f9488831-1348-4596-8eaf-4cb7b881e481.png)

**Comment by MWHunter** [comment]
closing because you seem to have fixed it with https://github.com/retrooper/packetevents/commit/67eaa912525490df157071e66d73c1b9c3b4d9d7

**Comment by caoli5288** [comment]
> closing because you seem to have fixed it with [67eaa91](https://github.com/retrooper/packetevents/commit/67eaa912525490df157071e66d73c1b9c3b4d9d7)

It's fixed correctly by #474
